cmd/sink/main.go

```diff
@@ -5,29 +5,58 @@ package main
 
 import (
 	"context"
+	"database/sql"
 	"log"
 	"net/http"
 	"os"
+	"time"
 
 	ceOtelObs "github.com/cloudevents/sdk-go/observability/opentelemetry/v2/client"
 	cloudevents "github.com/cloudevents/sdk-go/v2"
 	ceClient "github.com/cloudevents/sdk-go/v2/client"
+	kaDatabase "github.com/kubearchive/kubearchive/pkg/database"
 	kaObservability "github.com/kubearchive/kubearchive/pkg/observability"
+	_ "github.com/lib/pq"
 	"go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp"
 )
 
-var logger = log.New(os.Stderr, "", log.LstdFlags|log.Lmicroseconds|log.LUTC)
+var (
+	logger         = log.New(os.Stderr, "", log.LstdFlags|log.Lmicroseconds|log.LUTC)
+	db     *sql.DB = nil
+)
```

Comment:
Don't use global variables, follow the same pattern than the API: create a type struct called `Server` (or `Sink` if you want) and have logger and db be two properties of that struct. Then `receive` would be a method of that struct and will get the database connection as `s.db` and the logger as `s.logger` instead of  just `db`. For reference:

https://github.com/kubearchive/kubearchive/blob/main/cmd/api/main.go#L44

This allows us for a better testing experience, in which we can initialize the `Sink` struct with a fake db connection on each test, instead on having a shared global variable that depends on the order of execution of the tests.

Reply:
I'll see if I can get that approach to work with the cloudevent sdk

Reply:
Ah, I didn't take that into account. Give it a go and if you find yourself spending too much, create a ticket and we handle it next.

Reply:
it works! the cloudevents sdk has no issue calling a function or a struct method and there's no more global variables